src/app/shell/item-comparators.ts

```diff
@@ -240,3 +349,69 @@ export function sortItems(
   );
   return items.toSorted(comparator);
 }
+
+export function groupItems(
+  items: readonly DimItem[],
+  vaultGrouping: string,
+  getTag: (item: DimItem) => TagValue | undefined,
+): readonly VaultGroup[] {
+  const groupingValueGetter = GROUPING_VALUE_GETTERS[vaultGrouping];
+  const iconValueGetter = ICON_VALUE_GETTERS[vaultGrouping];
+  const comparator = GROUP_BY_COMPARATORS[vaultGrouping];
```

Comment:
Rather than have three parallel maps, you could have one map where each entry is an object consisting of value, icon, and comparator

Reply:
Yeah, will do - saves on some logic.